### PR TITLE
vendor: update buildkit to aaff9d591

### LINKED ISSUE #34708: Use official `moby/buildkit` in vendor
we're currently using a fork for the logrus fixes.

### PR DESCRIPTION
**Pull request by tonistiigi**
fixes #34708

### CODE REVIEW

File: builder/fscache/fscache.go [hunk 1]
```diff
@@ -607,3 +616,40 @@ func (s sortableCacheSources) Less(i, j int) bool {
 func (s sortableCacheSources) Swap(i, j int) {
 	s[i], s[j] = s[j], s[i]
 }
+
+func newTarsumHash(stat *fsutil.Stat) (hash.Hash, error) {
```
Comment (tonistiigi) [inline builder/fscache/fscache.go]: this code was previously in https://github.com/moby/moby/pull/34943/files#diff-3b65b3cb4440e2f8d672b27e896f75a8L314

File: builder/fscache/fscache.go [hunk 2]
```diff
@@ -607,3 +616,40 @@ func (s sortableCacheSources) Less(i, j int) bool {
...
+	fi := &fsutil.StatInfo{stat}
+	p := stat.Path
+	if fi.IsDir() {
+		p += string(os.PathSeparator)
```
Comment (dnephin) [inline builder/fscache/fscache.go]: Should this try to trim any trailing os.PathSeprator first in case it already has the suffix?

**Review by dnephin — APPROVED** [review]
LGTM

File: builder/fscache/fscache.go (hunk 2 above)
Comment (tonistiigi) [reply builder/fscache/fscache.go]: These paths are already cleaned https://github.com/moby/moby/pull/34943/files/d4729749023fde0c57cded0c8159dc85cd7ee448#diff-78afa8e80f1f2221c7bf7bc880b54f70R33

**Review by AkihiroSuda — APPROVED** [review]
LGTM if green

**Review by vdemeester — APPROVED** [review]
LGTM 🐮

**Pull request merged**